addons/event_crm/models/crm_lead.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+
+
+class Lead(models.Model):
+    _inherit = 'crm.lead'
+
+    event_id = fields.Many2one('event.event', string="Event linked to the lead")
+
+    registration_ids = fields.Many2many('event.registration', string="Registrations", help="Registrations linked to the lead")
```

Comment:
It is strange to have a m2m. After some thoughts, I suppose it comes from having multiple rules creating multiple leads ? Is it what we expect ?